sql/core/src/test/scala/org/apache/spark/sql/SparkSessionExtensionSuite.scala

```diff
@@ -1161,3 +1177,12 @@ object AddLimit extends Rule[LogicalPlan] {
     case _ => Limit(Literal(1), plan)
   }
 }
+
+object FailWithAQEShuffleRead extends Rule[SparkPlan] {
```

Comment:
Do you think we can have some meaningful rule?

Reply:
OK, thinking a new rule `RequireAtLeaseTwoPartitions` that would revert `AQEShuffleReadExec` if it only has one partition.